### ISSUE TITLE
"command not found" in bash_completion step

### ISSUE THREAD

**Issue by subsetpark** [scopatz/xonsh#682]
I recently installed [taskwarrior](taskwarrior.org), and now xonsh produces the following errors on startup:

```
/usr/local/etc/bash_completion.d/task.sh: line 75: task: command not found
/usr/local/etc/bash_completion.d/task.sh: line 75: task: command not found
~ $
```

However `task` is found normally from within the shell.

My path is set on the first line of my `~/.xonshrc`. So my guess is, there is some bash_completion stage in xonsh startup that is running before my PATH gets updated, and failing to find an executable in `/usr/local/bin`. Just as an experiment, I echoed my $PATH line into a new file called `/etc/xonshrc` but the same error message is displayed.

**Comment by scopatz** [comment]
Hi @subsetpark, did you add this file to your `$BASH_COMPLETIONS`?  Just a guess...

**Comment by subsetpark** [comment]
@scopatz - yes, it's there. It's the default:

```
['/usr/local/etc/bash_completion',
 '/opt/local/etc/profile.d/bash_completion.sh']
```

And so the script in `bash_completion.d` is being caught and attempting to execute.

**Comment by scopatz** [comment]
Well, $PATH is and the xonshrc file is executed before the completer is even instantiated: https://github.com/scopatz/xonsh/blob/master/xonsh/base_shell.py#L108

So this is some other issue, probably related to the fact that the environment isn't being passed in to the completion functions: https://github.com/scopatz/xonsh/blob/master/xonsh/completer.py#L466 and https://github.com/scopatz/xonsh/blob/master/xonsh/completer.py#L487

That should be a relatively quick fix.

**Comment by scopatz** [comment]
Note that ideally this stuff would be refactored to use the `foreign_shell` interface.

**Comment by scopatz** [comment]
Sorry about taking so long to get to this @subsetpark.  I put in a potential fix at #691. Would you mind trying it out?
